packages/headless/src/api/search/search-api-client.ts

```diff
@@ -124,25 +125,26 @@ export class SearchAPIClient implements FacetSearchAPIClient {
     };
   }
 
-  private searchAbortController: AbortController | null = null;
+  private searchAbortControllers: Partial<
```

Comment:
A refactor opportunity would be to create an `APIClientAbortController` class, which could hold the state implemented here, a constructor which would wrap `createAbortController` function, and, a reference to the `PlatformClient.call` function which could then automatically `clear` itself once the request finishes.

What do you think ?

Reply:
Yeah, I thought about making a dedicated class, this feels pretty intuitive to abstract and would look kinda clean, but I'm not sure if it would actually simplify the code that much. I also feel like it's one more thing we'd have to step through while debugging. It does shorten the file a bit though, and I like reducing visual pollution like this.

I'm open to changing it, I'm curious what others think too.

Reply:
Refactored into APICallsQueue